ms-gateway/src/main/java/net/trajano/ms/gateway/providers/RefreshHandler.java

```diff
@@ -0,0 +1,128 @@
+package net.trajano.ms.gateway.providers;
+
+import static io.vertx.core.http.HttpHeaders.DATE;
+import static java.time.ZoneOffset.UTC;
+import static java.time.ZonedDateTime.now;
+import static java.time.format.DateTimeFormatter.RFC_1123_DATE_TIME;
+import static net.trajano.ms.gateway.providers.RequestIDProvider.REQUEST_ID;
+
+import java.net.URI;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+import net.trajano.ms.gateway.internal.Conversions;
+import net.trajano.ms.gateway.internal.MediaTypes;
+
+/**
+ * This handler deals with refreshing the OAuth token.
+ */
+@Component
+public class RefreshHandler implements
+    Handler<RoutingContext> {
+
+    /**
+     * This is the pattern for the refresh token used.
+     */
+    private static final String TOKEN_PATTERN = "^[A-Za-z0-9]{64}$";
+
+    @Value("${authorization.endpoint}")
+    private URI authorizationEndpoint;
+
+    @Autowired
+    private HttpClient httpClient;
+
+    @Autowired
+    private RequestIDProvider requestIDProvider;
+
+    @Override
+    public void handle(final RoutingContext context) {
+
+        final HttpServerRequest contextRequest = context.request();
+        final HttpServerResponse contextResponse = context.response();
+
+        final String requestID = requestIDProvider.newRequestID(context);
+
+        final String grantType = contextRequest.getFormAttribute("grant_type");
+        if (grantType == null) {
+            contextResponse
+                .setChunked(false)
+                .putHeader(HttpHeaders.CONTENT_TYPE, MediaTypes.APPLICATION_JSON)
+                .setStatusCode(400)
+                .setStatusMessage("Bad Request")
+                .end(new JsonObject()
+                    .put("error", "invalid_grant")
+                    .put("error_description", "Missing grant type")
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "error_description" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)